### DISCUSSION TITLE
Workflow process in pimcore portal engine

### DISCUSSION THREAD

**Discussion by nirbanc2020** [pimcore/pimcore#9617] [Q&A / Support]
Hi,
I already setup the pimcore portal engine and created the portal users to access the portal.
I want to map role with portal users and in workflow I want to give access transitions though guard events to such role users.
I already tried such guard events in workflow.yml like
```  is_granted('ROLE_REVIEWER') ```
```is_fully_authenticated()``` but it is not working while is_granted ```('ROLE_PIMCORE_ADMIN')``` is working fine.
Please help regarding the same.

**Comment by fashxp** [comment]
Hi,
I'm not sure if that is possible so easily. 

Probably the easiest would be to listen for [workflow events](https://symfony.com/doc/current/workflow.html#using-events) and do your checks there. 

Otherwise you might need to decorate this [Pimcore service](https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/lib/Workflow/ExpressionService.php#L60) and inject the portal engine usernames and roles there. 


Please share your findings...

BR
Christian

**Reply by esorthapa** [reply]
Hi @nirbanc2020 and @fashxp 

I already had setup the pimcore portal and created the portal users too. 
I try to create Portal Setting using Portal Engine portal setup wizard but, I could not create Portal Setting with error lazy loading   and popup message with "portal creation failed". 

Also, found console with error on 
1>pimcore_direct_edit:
https://example.com/admin/pimcore_direct_edit/settings?_dc=1633431790 
with response: 500 Internal Server Error

2>Next with console message:
Admin user has no assigned portal user.

![portal_error](https://user-images.githubusercontent.com/29518403/136011390-dd89144f-fb9f-42b1-af98-4312927778ba.png)
![message](https://user-images.githubusercontent.com/29518403/136011410-f60c5d08-f080-48b3-9412-788549d51085.png)

Please suggest appropriate answer as soon as possible. Next, In case Domain, Do we need to add newly created domain or same domain as project url in "domain" in Portal setting form. Please provide me with sample.

**Comment by shakya-nischal** [comment]
> Hi, I already setup the pimcore portal engine and created the portal users to access the portal. I want to map role with portal users and in workflow I want to give access transitions though guard events to such role users. I already tried such guard events in workflow.yml like ` is_granted('ROLE_REVIEWER')` `is_fully_authenticated()` but it is not working while is_granted `('ROLE_PIMCORE_ADMIN')` is working fine. Please help regarding the same.

Hello @nirbanc2020 I am trying to install the portal engine on my local. My pimcore version is 6.9, pimcore/portal-engine version is 2.1 and my elasticsearch version is 7.17.  
Enterprise Metadata Bundle, statistics explorer and PimcorePortalEngineBundle is enabled.
When I try to run **./bin/console portal-engine:update:index-recreate** command it gives 
**In StaticNoPingConnectionPool.php line 67:**
**No alive nodes found in your cluster**
My elasticsearch is actively running

**In config.yml configured parameters:**
`parameters: pimcore_portal_engine.elasticsearch.host: 'localhost:9200' pimcore_portal_engine.elasticsearch.index_prefix: 'pimcore_portal_engine_elasticsearch_local'`

**In security.yml**
```
security:
firewalls:
pimcore_admin:
      # ...
  portal_engine: '%pimcore_portal_engine.firewall_settings%'
```

Can you help me through this problem!!

**Comment by esorthapa** [comment]
Hi shakes,

Using workflow and it’s yml guard event, transition must work.

I don’t how your full code is working.

Role_pimcore_admin worked as you mention, similarly if transitions are fine
other flow must work.

If any more help please let me know and share your workflow if possible.
